Mage/src/main/java/mage/players/PlayerImpl.java

```diff
@@ -2972,6 +2972,40 @@ public boolean seekCard(FilterCard filter, Ability source, Game game) {
         return true;
     }
 
+    @Override
+    public Set<Card> seekCard(FilterCard filter, Zone inZone, boolean tapped, int amount, Ability source, Game game) {
+        Player controller = game.getPlayer(source.getControllerId());
```

Comment:
I'm confused, why do you need controller? this is a method in PlayerImpl, so it seems like a redundant param. (If it's not dependent on the player, then it should instead be directly in the apply method of the Effect.)

Consider using PutCards rather than Zone + tapped.

Also, you should be able to simplify the existing seekCard method to call this more general one - don't want to have duplicate code.